### PR TITLE
Game window capture error handling

### PR DESCRIPTION
**Pull request by avoitenko-logitech**

### CODE REVIEW

**Comment by gettinToasty** [comment]
Looks like there was a strict null error on CI here

**Review by gettinToasty — CHANGES_REQUESTED** [review]
Tests need to be fixed here

**Comment by avoitenko-logitech** [comment]
`media-backup` test is failing which is not related to my changes.

File: app/services/sources/sources.ts [hunk 1]
```diff
@@ -356,7 +358,7 @@ export class SourcesService extends StatefulService<ISourcesState> {
     const source = this.views.getSource(id)!;
     const muted = obsInput.muted;
     this.UPDATE_SOURCE({ id, muted });
-    this.updateSourceFlags(source.state, obsInput.outputFlags, true);
+    this.updateSourceFlags(source, obsInput.outputFlags, true);
```
Comment (michelinewu) [inline app/services/sources/sources.ts:361]: Why remove `.state`?

File: app/services/sources/sources.ts [hunk 2]
```diff
@@ -616,16 +618,28 @@ export class SourcesService extends StatefulService<ISourcesState> {
     });
   }
 
-  private updateSourceFlags(source: ISource, flags: number, doNotEmit?: boolean) {
+  private updateSourceFlags(source: Source, flags: number, doNotEmit?: boolean) {
```
Comment (michelinewu) [inline app/services/sources/sources.ts:621]: Why `Source` instead of `ISource`?

Comment (avoitenko-logitech) [reply app/services/sources/sources.ts:621]: Because `ISource` does not have `getModel` method.

File: app/services/sources/sources.ts (hunk 1 above)
Comment (avoitenko-logitech) [reply app/services/sources/sources.ts:361]: Because I need to pass `Source` instead of `ISource`

**Pull request merged**